LibreNMS/Alert/Transport/Jira.php

```diff
@@ -29,54 +22,77 @@
 
 class Jira extends Transport
 {
+    protected string $name = 'Jira';
+
     public function deliverAlert(array $alert_data): bool
     {
-        // Don't create tickets for resolutions
-        if ($alert_data['severity'] == 'recovery') {
-            return true;
+        $webhook_on = $this->config['enable-webhook'];
+
+        // Check if messsage is an alert or not
+        if ($alert_data['state'] != 0) {
+            $url = $this->config['jira-url'];
+            // If webhooks are not enabled, append the API info
+            if (! $webhook_on) {
+                $url .= '/rest/api/latest/issue';
+            }
+            // Messsage is a recovery
+        } else {
+            if (! $webhook_on) {
+                return false; // Webhooks not enabled, do nothing.
+            } else {
+                $url = $this->config['jira-close-url'];
+            }
         }
 
-        $prjkey = $this->config['jira-key'];
-        $issuetype = $this->config['jira-type'];
-        $details = empty($alert_data['title']) ? 'Librenms alert for: ' . $alert_data['hostname'] : $alert_data['title'];
+        $project_key = $this->config['jira-key'];
+        $issue_type = $this->config['jira-type'];
+        $title = empty($alert_data['title']) ? 'Librenms alert for: ' . $alert_data['hostname'] : $alert_data['title'];
         $description = $alert_data['msg'];
-        $url = $this->config['jira-url'] . '/rest/api/latest/issue';
 
+        // Construct the payload
         $data = [
             'fields' => [
+                'summary' => $title,
+                'description' => $description,
                 'project' => [
-                    'key' => $prjkey,
+                    'key' => $project_key,
                 ],
-                'summary' => $details,
-                'description' => $description,
                 'issuetype' => [
-                    'name' => $issuetype,
+                    'name' => $issue_type,
                 ],
             ],
         ];
 
+        // Add Custom Webhook ID to the payload
+        if ($webhook_on) {
+            if (! empty($this->config['webhook-id'])) {
```

Comment:
You need to check that ` $data['fields'][$this->config['webhook-id']]` is set too.

Reply:
Perhaps throwing an exception if it is not set (telling the user that the given field does not exist)

Reply:
ok I've set it to default key to "alert_id" if value is not set, simpler if you don't need to specify your ID